### PR TITLE
Add simpleConsistency and getMeta methods to API

### PR DESCRIPTION
**Pull request by JettTech**
### Updates:
 - relocates consistency logic to a single file
 - adds simpleConsistency to api
 - adds getMeta to api
 - updates 'late joiners' consistency test to consume simpleConsistency method

### CODE REVIEW

**Review by zo-el — APPROVED** [review]

File: src/api.ts [hunk 1]
```diff
@@ -164,6 +165,21 @@ export class ScenarioApi {
...
+   simpleConsistency = async (instance_id, players: Array<Player>): Promise<Boolean> => {
+      var retries = 3
+      while (!await isConsistent(instance_id, players)) {
+        retries--
+        if (retries == 0) {
+          return false
+        }
+        await delay(10000)
+      }
+      return true
```
Comment (maackle) [inline src/api.ts:178]: Is there harm in doing more retries with a smaller interval, to spend less time waiting? Or does isConsistent take a long time itself?

Comment (maackle) [reply src/api.ts:178]: If getting state dumps is fairly quick, I'd be fine with replacing `s.consistency` with this, so there is only one way of doing it.

**Review by maackle — APPROVED** [review]
Seems legit. We could consider just replacing `s.consistency` with the new way, although I'd like to see the same features, i.e. configurable timeout (and polling interval), and perhaps a strict/lax mode like in `s.consistency`

File: src/api.ts (hunk 1 above)
Comment (zippy) [reply src/api.ts:178]: The reason I didn't want to replace s.consistency is that going forward I didn't want people to believe that there was a generic way to determine consistency, that you actually have to write into your tests what that means to you if its anything beyond simple.

**Review by zippy — APPROVED** [review]

**Pull request merged**